eq-author-api/migrations/updateMetadataValue.test.js

```diff
@@ -0,0 +1,73 @@
+const updateMetadataValue = require("./updateMetadataValue.js");
+
+describe("updateMetadataValue", () => {
+  it("should update metadata to correct new format", () => {
+    const metadata = [
+      {
+        alias: "text",
+        dateValue: null,
+        id: "296",
+        key: "ru_ref",
+        languageValue: null,
+        regionValue: null,
+        textValue: "old Value",
+        type: "Text",
+        value: "new Value",
+      },
+      {
+        alias: "date",
+        dateValue: "2019-07-01",
+        id: "294",
+        key: "date",
+        languageValue: null,
+        regionValue: null,
+        textValue: null,
+        type: "Date",
+        value: "2019-04-09",
+      },
+      {
+        alias: "region",
+        dateValue: null,
+        id: "297",
+        key: "region",
+        languageValue: null,
+        regionValue: "GB_GBN",
+        textValue: null,
+        type: "Region",
+        value: "GB_ENG",
+      },
+      {
+        alias: "language",
+        dateValue: null,
+        id: "297",
+        key: "language",
+        languageValue: "cy",
+        regionValue: null,
+        textValue: null,
+        type: "Language",
+        value: "en",
+      },
+      {
+        alias: "language",
+        dateValue: null,
+        id: "297",
+        key: "language",
+        languageValue: "cy",
+        regionValue: null,
+        textValue: null,
+        type: "Region",
+      },
+    ];
+    const updatedQuestionnaire = updateMetadataValue({ metadata });
+    updatedQuestionnaire.metadata.forEach(metadata => {
+      expect(metadata).not.toHaveProperty("value");
+    });
+    expect(updatedQuestionnaire.metadata[0].textValue).toEqual("new Value");
+    expect(updatedQuestionnaire.metadata[1].dateValue).toEqual("2019-04-09");
+    expect(updatedQuestionnaire.metadata[2].regionValue).toEqual("GB_ENG");
+    expect(updatedQuestionnaire.metadata[3].languageValue).toEqual("en");
+    expect(updatedQuestionnaire.metadata[4].languageValue).toEqual(
```

Comment:
Shouldn't this be null given that the type is different. The metadata is completely wrong really as regionValue is also null

Reply:
Should it? Think I might be missing something here, this final expect tests that if there is no value field in the db the metadata is unchanged?

Reply:
@SamGodwin2 I think it might be worth making that explicit by having a separate test. I was confused by that last expect too.

e.g. `it("should not change existing values if no value field")`

Reply:
Ah yeah I see it now the metadata is completely wrong, should be a `Language` type not `Region` I swap that and move it into a seperate test for clarity.